pystac/extensions/eo.py

```diff
@@ -459,6 +459,35 @@ def _get_bands(self) -> Optional[List[Band]]:
             return [Band(b) for b in bands]
         return None
 
+    def get_assets(
+        self,
+        name: Optional[str] = None,
+        common_name: Optional[str] = None,
+    ) -> Dict[str, pystac.Asset]:
+        """Get the item's assets where eo:bands are defined.
+
+        Args:
+            name: If set, filter the assets such that only those with a
+                matching ``eo:band.name`` are returned.
+            common_name: If set, filter the assets such that only those with a matching
+                ``eo:band.name`` are returned.
+
+        Returns:
+            Dict[str, Asset]: A dictionary of assets that match ``name``
+                and/or ``common_name`` if set or else all of this item's assets were
+                eo:bands are defined.
+        """
+        kwargs = {"name": name, "common_name": common_name}
+        return {
+            key: asset
+            for key, asset in self.item.get_assets().items()
+            if BANDS_PROP in asset.extra_fields
+            and all(
+                v is None or v in [b[k] for b in asset.extra_fields[BANDS_PROP]]
```

Comment:
What happens if the band doesn't have `name` or `common_name`? It _looks_ like it would `KeyError` but maybe I'm missing something.

Reply:
yeah it would KeyError, good call.